dirsrvtests/tests/suites/vlv/regression_test.py

```diff
@@ -246,6 +259,103 @@ def test_vlv_recreation_reindex(topology_st):
     check_vlv_search(conn)
 
 
+def verify_vlv_subdb_names(vlvname, inst):
+    """
+    Verify that vlv index and cache sub database names are conistent on lmdb.
+    """
+    if get_default_db_lib() != "mdb":
+        return
+    output = run_dbscan(['-D', 'mdb', '-L', inst.dbdir])
+    vlvsubdbs = []
+    nbvlvindex = 0
+    for line in output.split("\n"):
+        if not line:
+            continue
+        fname = line.split()[0]
+        if 'vlv#' in fname:
+            vlvsubdbs.append(fname)
+            if '/~recno-cache/' not in fname:
+                nbvlvindex += 1
+    log.info(f'vlv sub databases are: {vlvsubdbs}')
+    assert f'{inst.dbdir}/userroot/vlv#{vlvname}.db' in vlvsubdbs
+    assert f'{inst.dbdir}/userroot/~recno-cache/vlv#{vlvname}.db' in vlvsubdbs
+    assert len(vlvsubdbs) == 2 * nbvlvindex
+
+
+def test_vlv_cache_subdb_names(topology_m2):
+    """
+    Testing that vlv index cache sub database name is consistent.
+
+    :id: 7a138006-aa24-11ee-9d5f-482ae39447e5
+    :setup: Replication with two suppliers.
+    :steps:
+        1. Generate vlvSearch and vlvIndex entries
+        2. Add users and wait that they get replicated.
+        3. Check subdb names and perform vlv searches
+        4. Reinit M2 from M1
+        5. Check subdb names and perform vlv searches
+    :expectedresults:
+        1. Should Success.
+        2. Should Success.
+        3. Should Success.
+        4. Should Success.
+        5. Should Success.
+    """
+
+    NUM_USERS = 50
+    M1 = topology_m2.ms["supplier1"]
+    M2 = topology_m2.ms["supplier2"]
+    repl = ReplicationManager(DEFAULT_SUFFIX)
+
+    # Clean test_vlv_recreation_reindex leftover that cause trouble
+    entries = [ UserAccount(M1, dn=f'uid=testuser{uid},DEFAULT_SUFFIX') for uid in
+        range(STARTING_UID_INDEX, STARTING_UID_INDEX+NUMUSERS_TEST_VLV_REINDEX) ]
+    entries.append(VLVIndex(M2, dn='cn=vlvIdx,cn=vlvSrch,cn=userRoot,cn=ldbm database,cn=plugins,cn=config'))
+    entries.append(VLVSearch(M2, dn='cn=vlvSrch,cn=userRoot,cn=ldbm database,cn=plugins,cn=config'))
+    for entry in entries:
+        if entry.exists():
+            entry.delete()
+    repl.wait_for_replication(M1, M2)
+    # Restart the instance to workaround github issue #6028
+    M2.restart()
```

Comment:
If I understood correctly, the workaround is required only for BDB. Should we restart only for BDB tests, then? (by adding some if here or something)

Also, kind of nitpick, I think it'll be useful to have a full link here (`https://github.com/389ds/389-ds-base/issues/6028`) in case we'll change the bug tracker somewhere in the future.

Reply:
The new testcase is still failing, and I realized that I made a mistake when creating issue 6028 
  (used wrong test case in the reproducer) 
Redoing the test I noticed that the inconsistency occurs only in mdb and the hang only in bdb 
So I fixed 6028 description for mdb inconsistency and created #6029 for the bdb hang.
I changed the code to do as you suggested.
  use full link to  https://github.com/389ds/389-ds-base/issues/6029 and restart M2 only in bdb case
  
Note the new test case is still failing with mdb but fixing  6028 should solve it.